File: server/router/task/daily_task_completion.go
package task

import (
	"github.com/gin-gonic/gin"
	"github.com/springbear2020/self-hub/server/middleware"
)

type DailyTaskCompletionRouter struct{}

func (s *DailyTaskCompletionRouter) InitDailyTaskCompletionRouter(Router *gin.RouterGroup, PublicRouter *gin.RouterGroup) {
	taskCompletionRouter := Router.Group("dailyTaskCompletion").Use(middleware.OperationRecord())
	{
		taskCompletionRouter.POST("createDailyTaskCompletion", dailyTaskCompletionApi.CreateDailyTaskCompletion)
		taskCompletionRouter.DELETE("deleteDailyTaskCompletion", dailyTaskCompletionApi.DeleteDailyTaskCompletion)
		taskCompletionRouter.DELETE("deleteDailyTaskCompletionByIds", dailyTaskCompletionApi.DeleteDailyTaskCompletionByIds)
		taskCompletionRouter.PUT("updateDailyTaskCompletion", dailyTaskCompletionApi.UpdateDailyTaskCompletion)
	}

	taskCompletionRouterWithoutRecord := Router.Group("dailyTaskCompletion")
	{
		taskCompletionRouterWithoutRecord.GET("findDailyTaskCompletion", dailyTaskCompletionApi.FindDailyTaskCompletion)
		taskCompletionRouterWithoutRecord.GET("getDailyTaskCompletionList", dailyTaskCompletionApi.GetDailyTaskCompletionList)
		taskCompletionRouterWithoutRecord.GET("getDailyTaskStatList", dailyTaskCompletionApi.GetDailyTaskStatList)
	}

	taskCompletionRouterWithoutAuth := PublicRouter.Group("dailyTaskCompletion")
	{
		_ = taskCompletionRouterWithoutAuth
	}
}

File: web/src/api/task/daily_task_completion.js
import service from '@/utils/request'

export const createDailyTaskCompletion = (data) => {
  return service({
    url: '/dailyTaskCompletion/createDailyTaskCompletion',
    method: 'post',
    data
  })
}

export const deleteDailyTaskCompletion = (params) => {
  return service({
    url: '/dailyTaskCompletion/deleteDailyTaskCompletion',
    method: 'delete',
    params
  })
}

export const deleteDailyTaskCompletionByIds = (params) => {
  return service({
    url: '/dailyTaskCompletion/deleteDailyTaskCompletionByIds',
    method: 'delete',
    params
  })
}

export const updateDailyTaskCompletion = (data) => {
  return service({
    url: '/dailyTaskCompletion/updateDailyTaskCompletion',
    method: 'put',
    data
  })
}

export const findDailyTaskCompletion = (params) => {
  return service({
    url: '/dailyTaskCompletion/findDailyTaskCompletion',
    method: 'get',
    params
  })
}

export const getDailyTaskCompletionList = (params) => {
  return service({
    url: '/dailyTaskCompletion/getDailyTaskCompletionList',
    method: 'get',
    params
  })
}

export const getDailyTaskStatList = (params) => {
  return service({
    url: '/dailyTaskCompletion/getDailyTaskStatList',
    method: 'get',
    params
  })
}
